File: modules/module-bookkeeping/src/main/kotlin/ch/awae/mycloud/module/bookkeping/model/AccountType.kt
package ch.awae.mycloud.module.bookkeping.model

enum class AccountType(val shortString: String, val invertedPresentation: Boolean?) {
    ASSET("A", false),
    LIABILITY("P", true),
    TRANSFER("X", null),
    EXPENSE("-", false),
    INCOME("+", true),
    ;

    val earningsAccount: Boolean
        get() = this == EXPENSE || this == INCOME

}

File: services/angular-frontend/src/app/apps/bookkeeping/model/accountType.ts
export enum AccountType {
  ASSET = "ASSET",
  LIABILITY = "LIABILITY",
  EXPENSE = "EXPENSE",
  INCOME = "INCOME",
}

export class AccountTypeUtil {
  public static iconForType(type: AccountType): string {
    switch (type) {
      case AccountType.ASSET:
        return 'savings'
      case AccountType.LIABILITY:
        return 'credit_card'
      case AccountType.INCOME:
        return 'trending_up'
      case AccountType.EXPENSE:
        return 'trending_down'
    }
  }
}
